ese-varo/growing
# Feature/habit checkpoints

## Explain why this change is being made.
As a registered user with active habits, I want to see the checkpoints of a specific day of the habit, so that I can see all the checkpoints about the habit.

## Explain how this is accomplished.
Once a habit is created, you could see the list of the days duration of the habit, you could press over each day to add, edit o remove a checkpoint

## How do you manually test this?
  - Given the user is logged in, you can press the link Habits over the navbar, then click on 'Add new habit', fill all the form, and once the habit is created you will be redirected to the details page of the habit
  - Once you see the habit details page if you click on any day of the calendar a card with information about the checkpoint for that day is displayed, if there is an existing checkpoint for that day it will allow you to edit it or remove it, if not, an option for adding one would be available.

### Screenshots
- Create Checkpoint
![image](https://user-images.githubusercontent.com/31193448/114437483-39a87480-9b8c-11eb-84ab-2aa99cd9860b.png)
![image](https://user-images.githubusercontent.com/31193448/114437552-504ecb80-9b8c-11eb-9fbd-afd1b14f5520.png)
![image](https://user-images.githubusercontent.com/31193448/114437597-6066ab00-9b8c-11eb-942d-659c433eeb20.png)
- Edit
![image](https://user-images.githubusercontent.com/31193448/114437737-8f7d1c80-9b8c-11eb-9b23-a83645ab8cd5.png)
![image](https://user-images.githubusercontent.com/31193448/114437799-9f94fc00-9b8c-11eb-9b75-968e90e48f8c.png)

- Delete
![image](https://user-images.githubusercontent.com/31193448/114437866-b4718f80-9b8c-11eb-8364-5d302bcf5edb.png)
![image](https://user-images.githubusercontent.com/31193448/114437926-c5ba9c00-9b8c-11eb-86e5-5050a80c1c88.png)
